pkg/test-fw/ig/ig.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ig
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/blang/semver"
+)
+
+type IG struct {
+	path    string
+	image   string
+	version semver.Version
+
+	cmd    *exec.Cmd
+	env    []string
+	flags  []string
+	stdout bytes.Buffer
+	stderr bytes.Buffer
+
+	// StartAndStop indicates this command should first be started then stopped.
+	// It corresponds to gadget like execsnoop which wait user to type Ctrl^C.
+	StartAndStop bool
+
+	// started indicates this command was started.
+	// It is only used by command which have StartAndStop set.
+	started bool
+
+	// Cleanup indicates this command is used to clean resource and should not be
+	// skipped even if previous commands failed.
+	Cleanup bool
+}
+
+func (ig *IG) IsCleanup() bool {
+	return ig.Cleanup
+}
+
+func (ig *IG) IsStartAndStop() bool {
+	return ig.StartAndStop
+}
+
+func (ig *IG) Running() bool {
+	return ig.started
+}
+
+func (ig *IG) createCmd() {
+	args := append([]string{"run", ig.image}, ig.flags...)
+	cmd := exec.Command(ig.path, args...)
+
+	ig.env = append(ig.env, "IG_EXPERIMENTAL=true")
+	cmd.Env = append(cmd.Env, ig.env...)
+
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = &ig.stdout
+	cmd.Stderr = &ig.stderr
+
+	// To be able to kill the process of ig.cmd,
+	// we need to send the termination signal to their process group ID
+	// (PGID). However, child processes get the same PGID as their parents by
+	// default, so in order to avoid killing also the integration tests process,
+	// we set the fields Setpgid and Pgid of syscall.SysProcAttr before
+	// executing ig.cmd. Doing so, the PGID of ig.cmd (and its children)
+	// will be set to its process ID, see:
+	// https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/syscall/exec_linux.go;l=32-34.
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true, Pgid: 0}
+
+	ig.cmd = cmd
+}
+
+func (ig *IG) run() error {
+	ig.createCmd()
+	if err := ig.cmd.Run(); err != nil {
+		switch e := err.(type) {
+		case *exec.Error:
+			return fmt.Errorf("command execution: %w", err)
+		case *exec.ExitError:
+			return fmt.Errorf("command exit code = %d", e.ExitCode())
+		default:
+			return err
+		}
```

Comment:
Do you really need this switch statement here? Isn't it possible to return `err`?